### ISSUE TITLE
Deleting a namespace in Kubernetes does not delete the corresponding environment in rancher

### ISSUE THREAD

**Issue by soumyalj** [rancher/rancher#5508]
**Rancher Version:**:v1.2.0-pre1-rc1

**Docker Version:**1.10.3

**OS and where are the hosts located? (cloud, bare metal, etc):**: Ubuntu, Cloud

**Environment Type: (Cattle/Kubernetes/Swarm/Mesos)**: Kubernetes

**Steps to Reproduce:**
1. Create a Kubernetes environment(1.3) using https://github.com/rancher/qa-catalog/
2. Create a namespace e.g. "testingress10" and pods within in
3. Delete the namespace

**Results:** 
The corresponding environment in rancher does not get deleted.

mysql> select id, name, state, created, removed from environment;
```
+----+------------------------+--------+---------------------+---------+
| id | name                   | state  | created             | removed |
+----+------------------------+--------+---------------------+---------+
|  1 | Kubernetes             | active | 2016-07-20 22:57:49 | NULL    |
|  2 | default                | active | 2016-07-20 23:00:27 | NULL    |
|  3 | kubernetes-ingress-lbs | active | 2016-07-20 23:00:50 | NULL    |
|  4 | testingress10          | active | 2016-07-21 17:46:26 | NULL    |
|  5 | testingress10          | active | 2016-07-21 18:02:30 | NULL    |
|  6 | testingress10          | active | 2016-07-21 18:45:52 | NULL    |
|  7 | testingress10          | active | 2016-07-21 19:00:07 | NULL    |
|  8 | testingress10          | active | 2016-07-21 19:10:28 | NULL    |
|  9 | testingress10          | active | 2016-07-21 19:20:55 | NULL    |
| 10 | testingress10          | active | 2016-07-21 19:24:26 | NULL    |
| 11 | testingress10          | active | 2016-07-21 19:36:34 | NULL    |
| 12 | testingress2           | active | 2016-07-21 20:23:11 | NULL    |
| 13 | testingress15          | active | 2016-07-21 20:32:14 | NULL    |
| 14 | testingress15          | active | 2016-07-21 20:34:47 | NULL    |
| 15 | testingress15          | active | 2016-07-21 20:45:23 | NULL    |
| 16 | testingress9           | active | 2016-07-21 20:55:10 | NULL    |
| 17 | testingress9           | active | 2016-07-21 20:59:11 | NULL    |
| 18 | testingress15          | active | 2016-07-21 21:07:15 | NULL    |
| 19 | testingress15          | active | 2016-07-21 21:21:05 | NULL    |
+----+------------------------+--------+---------------------+---------+
```
Note: testingress10, testingress15, testingress9 namespaces have been deleted through kubectl, but the corresponding environments have not been deleted in rancher.

**Comment by alena1108** [comment]
@wlan0 

there are going to be 2 fixes: one for ingress to rely on the stack external_id when figure out the service - I'll work on it, it will work around the issue for ingress at least. This fix would go to this release.

as the future fix  - we should listen on namespace event in rancher kubernetes-agent, and generate corresponding rancher event (agent communicates with rancher via events). That can go to the future release.

**Comment by wlan0** [comment]
Fixed by https://github.com/rancher/kubernetes-agent/pull/28 and https://github.com/rancher/cattle/pull/1854

**Comment by sangeethah** [comment]
@wlan0 , This will be a good fix to go in for k8s 1.2 as well.

Current kubernetes agent we have is - rancher/kubernetes-agent:v0.2.1. It dont think it has the fix -  rancher/kubernetes-agent#28. Can you provide a K8s template with the latest agent ?

**Comment by soumyalj** [comment]
Verified using v1.1.4-rc1.  The following is the database query before and after deletion

```
mysql>  select id, name, state, created, removed from environment;
+----+--------------------------+--------+---------------------+---------+
| id | name                     | state  | created             | removed |
+----+--------------------------+--------+---------------------+---------+
|  1 | Kubernetes               | active | 2016-09-14 19:11:24 | NULL    |
|  2 | default                  | active | 2016-09-14 19:13:49 | NULL    |
|  3 | kubernetes-ingress-lbs   | active | 2016-09-14 19:14:05 | NULL    |
|  4 | testingress6             | active | 2016-09-14 19:19:38 | NULL    |
|  5 | kubernetes-loadbalancers | active | 2016-09-14 20:08:30 | NULL    |
+----+--------------------------+--------+---------------------+---------+
5 rows in set (0.00 sec)

mysql>  select id, name, state, created, removed from environment;
+----+--------------------------+---------+---------------------+---------------------+
| id | name                     | state   | created             | removed             |
+----+--------------------------+---------+---------------------+---------------------+
|  1 | Kubernetes               | active  | 2016-09-14 19:11:24 | NULL                |
|  2 | default                  | active  | 2016-09-14 19:13:49 | NULL                |
|  3 | kubernetes-ingress-lbs   | active  | 2016-09-14 19:14:05 | NULL                |
|  4 | testingress6             | removed | 2016-09-14 19:19:38 | 2016-09-14 23:15:04 |
|  5 | kubernetes-loadbalancers | active  | 2016-09-14 20:08:30 | NULL                |
+----+--------------------------+---------+---------------------+---------------------+
5 rows in set (0.00 sec)




```
